awx/main/fields.py

```diff
@@ -256,20 +268,18 @@ def bind_m2m_changed(self, _self, _role_class, cls):
             field_names = [field_names]
 
         for field_name in field_names:
-            # Handle the OR syntax for role parents
-            if type(field_name) == tuple:
-                continue
-
-            if type(field_name) == bytes:
-                field_name = field_name.decode('utf-8')
```

Comment:
Background here:

I added a new test to set expectations. We don't allow tuples in these entries (or bytes). The test ensures we don't have that anywhere. If someone adds support for it later, that's fine. But until that happens, I want an alert raised if the entries are not an expected structure, and the _last_ thing I want is a `continue`.